src/pages/~/[id].astro

```diff
@@ -4,72 +4,155 @@ import { firestore, getGame, getSession } from '../../lib/game-saving/account'
 import StandardHead from '../../components/standard-head.astro'
 import { signal } from '@preact/signals'
-import type { PersistenceState } from '../../lib/state'
+import { ConnectionStatus, PersistenceStateKind, RoomState, type PersistenceState } from '../../lib/state'
 import MobileUnsupported from '../../components/popups-etc/mobile-unsupported'
 import { mobileUserAgent } from '../../lib/utils/mobile'
+import { isAccountWhitelistedToUseCollabAndSavingBetaFeatures } from '../../lib/game-saving/account'
 
 const session = await getSession(Astro.cookies)
 if (!session) return Astro.redirect(`/login?to=${Astro.request.url}`, 302)
 
-const game = await getGame(Astro.params.id!)
-if (!game || game.ownerId !== session.user.id) return Astro.redirect('/404', 302)
-
-const fileRegexp = /^.*\/(.+)-(\d+)\.md$/
-
-let tutorial: string[] | undefined;
-let tutorialIndex: number | undefined;
-if (game.tutorialName) {
-	const files = await Astro.glob('/games/*.md')
-		
-		tutorial = files.filter(file => {
-			const regexedFile = file.file.match(fileRegexp)
-			return regexedFile && regexedFile[1] === game.tutorialName
-		})
-		?.map(md => md.compiledContent())
-	tutorialIndex = game.tutorialIndex
-}
+let persistenceState;
+let isMobile;
+let roomState;
+
+if(await isAccountWhitelistedToUseCollabAndSavingBetaFeatures(session.user.id, session.user.email)){
+	let checkRoom = false;
+	const game = await getGame(Astro.params.id!)
+	if (!game) return Astro.redirect('/404', 302)
+	if(game.ownerId !== session.user.id) checkRoom = true;
+
+	const fileRegexp = /^.*\/(.+)-(\d+)\.md$/
 
-let _persistenceState: PersistenceState
-if (session.session.full) {
-	if (game.unprotected) {
-		await firestore.collection('games').doc(game.id).update({ unprotected: false })
-		game.unprotected = false
+	let tutorial: string[] | undefined;
+	let tutorialIndex: number | undefined;
+	if (game.tutorialName) {
+		const files = await Astro.glob('/games/*.md')
+			
+			tutorial = files.filter(file => {
+				const regexedFile = file.file.match(fileRegexp)
+				return regexedFile && regexedFile[1] === game.tutorialName
+			})
+			?.map(md => md.compiledContent())
+		tutorialIndex = game.tutorialIndex
 	}
-	if (Astro.cookies.get('sprigTempGame').value === game.id)
-		Astro.cookies.delete('sprigTempGame', { path: '/' })
-
-	_persistenceState = {
-		kind: 'PERSISTED',
-		session,
-		cloudSaveState: 'SAVED',
-		game,
-		tutorial,
-		tutorialIndex,
-		stale: false
+
+	let _persistenceState: PersistenceState
+	if(checkRoom){
+		_persistenceState = {
+			kind: PersistenceStateKind.COLLAB,
+			game: game.id,
+			password: undefined,
+			cloudSaveState: "SAVED",
+			session: session,
+			stale: false
+		}
+	} else {
+		if (session.session.full) {
+			if (game.unprotected) {
+				await firestore.collection('games').doc(game.id).update({ unprotected: false })
+				game.unprotected = false
+			}
+			if (Astro.cookies.get('sprigTempGame').value === game.id)
```

Comment:
add a comment to explain what is happening here and its significance.

Reply:
I'm not 100% sure, it was there before my changes